docs/concepts.rst

```diff
@@ -0,0 +1,128 @@
+Concepts
+########
+
+
+Basically, *Kinto* is a service where client applications can store and retrieve JSON data.
+
+In order to provide synchronisation and sharing features for these data, *Kinto*
+introduces some basic concepts.
+
+*Kinto* objects in brief :
+
++-----------------+---------------------------------------------------------+
+| Object          | Description                                             |
++=================+=========================================================+
+| **bucket**      | Buckets can be seen as namespaces:                      |
+|                 | collections names won't collide if stored in different  |
+|                 | buckets.                                                |
++-----------------+---------------------------------------------------------+
+| **collection**  | A collection of records                                 |
++-----------------+---------------------------------------------------------+
+| **record**      | The actual stored data                                  |
++-----------------+---------------------------------------------------------+
+| **group**       | A named list of :term:`principals <principal>`.         |
++-----------------+---------------------------------------------------------+
+
+
+.. _concepts-buckets-collections-records:
+
+Buckets, Collections and Records
+================================
+
+A **record** is the smallest unit of data. By default, there is no schema,
+and the JSON can contain anything.
+
+A **collection** is a group of records. Records are manipulated as a list
+and can be filtered or sorted. Clients can obtain the list of changes that
+occured on the collection records since a certain revision (e.g. *last synchronisation*).
+
+A **bucket** is an abstract notion used to organize collections and their
+permissions. A bucket named ``default`` is provided, whose collections and records
+are accessible to the current user only.
+
+.. image:: images/concepts-general.png
+
+Every kind of object manipulated by *Kinto* shares some common properties:
+
+* a unique identifier;
+* a revision number, automatically incremented on change;
+* a set of permissions.
+
+Those concepts are very similar to a hard disk, where buckets would be partitions,
+collections are folders and records are files!
+
+
+.. _concepts-groups:
+
+Groups
+======
+
+*Kinto* also provides the concept of user groups, in order to define permissions.
```

Comment:
Consider:
> Kinto has a concept of *groups* of users. A group has a list of members and belongs to a bucket.
Permissions can refer to the *group* instead of an individual user - this makes it easy to define "roles", especially if the same set of permissions is applied to several objects.